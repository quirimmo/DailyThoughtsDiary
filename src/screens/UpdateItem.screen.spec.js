import React from 'react';
import { Alert } from 'react-native';
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';
import renderer from 'react-test-renderer';
import UpdateItemScreen from './UpdateItem.screen';

const item = { id: 'item' };
const mockGetParam = jest.fn(() => item);
const navigation = {
	getParam: mockGetParam
};
let component;

describe('UpdateItemScreen', () => {
	beforeEach(() => {
		jest.useFakeTimers();
		NavigationTestUtils.resetInternalState();
		jest.clearAllMocks();
		component = renderer.create(<UpdateItemScreen navigation={navigation} />);
	});
	afterEach(() => {
		jest.clearAllMocks();
	});

	it('should be defined', () => {
		expect(UpdateItemScreen).toBeDefined();
	});

	it('should render the component', async () => {
		expect(component.toJSON()).toMatchSnapshot();
	});

	it('should call the getParam method of the navigation prop', () => {
		expect(mockGetParam).toHaveBeenCalledWith('item');
	});

	describe('state', () => {
		describe('currentItem', () => {
			it('should be defined', () => {
				expect(component.getInstance().state.currentItem).toEqual(item);
			});
		});
	});

	describe('onSubmitItem', () => {
		let instance;
		let mockItemProp;

		describe('success', () => {
			beforeEach(() => {
				mockItemProp = jest.fn(() => 'item updated correctly');
				instance = renderer
					.create(<UpdateItemScreen navigation={navigation} updateItem={mockItemProp} />)
					.getInstance();
			});
			it('should call the addItem prop', async () => {
				instance.onSubmitItem();
				expect(mockItemProp).toHaveBeenCalled();
			});

			it('should call the alert method of Alert for displaying the result', async () => {
				const spy = jest.spyOn(Alert, 'alert');
				await instance.onSubmitItem();
				expect(spy).toHaveBeenCalledWith('item updated correctly');
			});
		});

		describe('error', () => {
			beforeEach(() => {
				mockItemProp = jest.fn(() => {
					throw new Error('error updating the item');
				});
				instance = renderer
					.create(<UpdateItemScreen navigation={navigation} updateItem={mockItemProp} />)
					.getInstance();
			});
			it('should call the addItem prop', async () => {
				instance.onSubmitItem();
				expect(mockItemProp).toHaveBeenCalled();
			});

			it('should call the alert method of Alert for displaying the error', async () => {
				const spy = jest.spyOn(Alert, 'alert');
				await instance.onSubmitItem();
				expect(spy).toHaveBeenCalledWith(new Error('error updating the item'));
			});
		});
	});
});
