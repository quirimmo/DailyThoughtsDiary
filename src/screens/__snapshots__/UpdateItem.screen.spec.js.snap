// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdateItemScreen should render the component 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "padding": 10,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "marginBottom": 10,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={undefined}
      >
        Symptoms:
      </Text>
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Type symptoms..."
        style={
          Object {
            "fontSize": 16,
            "height": 50,
          }
        }
        value={undefined}
      />
    </View>
    <View
      style={
        Object {
          "marginBottom": 10,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={undefined}
      >
        Location:
      </Text>
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Type where you are..."
        style={
          Object {
            "fontSize": 16,
            "height": 50,
          }
        }
        value={undefined}
      />
    </View>
    <View
      style={
        Object {
          "marginBottom": 10,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={undefined}
      >
        Thoughts:
      </Text>
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Type thoughts..."
        style={
          Object {
            "fontSize": 16,
            "height": 50,
          }
        }
        value={undefined}
      />
    </View>
    <View
      style={
        Object {
          "marginBottom": 10,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 18,
          }
        }
      >
        Time:
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 0.5,
              "flexDirection": "row",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 16,
              }
            }
          >
            13:20
          </Text>
        </View>
        <View
          style={
            Object {
              "flex": 0.5,
            }
          }
        >
          <View
            accessibilityComponentType="button"
            accessibilityLabel={undefined}
            accessibilityTraits={
              Array [
                "button",
              ]
            }
            accessible={true}
            collapsable={undefined}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Array [
                  Object {},
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                disabled={undefined}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#007AFF",
                      "fontSize": 18,
                      "padding": 8,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                Change Time
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "marginTop": 30,
        }
      }
    >
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits={
          Array [
            "button",
            "disabled",
          ]
        }
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {},
              Object {},
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "#cdcdcd",
                },
              ]
            }
          >
            Update Item
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
